src/qt/diagnosticsdialog.cpp

```diff
@@ -325,7 +610,10 @@ void DiagnosticsDialog::clkFinished()
 {
     if (udpSocket->waitForReadyRead())
     {
-        while (udpSocket->hasPendingDatagrams())
+        int64_t start_time = GetAdjustedTime();
+
+        // Only allow this loop to run for 15 seconds maximum.
+        while (udpSocket->hasPendingDatagrams() && GetAdjustedTime() - start_time <= 5)
```

Comment:
Should `5` be `15`?

```suggestion
        // Only allow this loop to run for 15 seconds maximum.
        while (udpSocket->hasPendingDatagrams() && GetAdjustedTime() - start_time <= 15)
```

Reply:
I changed it from 15 to 5 and forgot to change the comment.